sklearn/model_selection/_split.py

```diff
@@ -602,6 +604,40 @@ def _iter_test_masks(self, X, y=None, labels=None):
         for i in range(self.n_folds):
             yield test_folds == i
 
+    def split(self, X, y=None, labels=None):
```

Comment:
actually `y` is not optional in a `StratifiedKFold`. The splits are dependent on `y`